src/NUnitFramework/framework/Internal/Commands/TimeoutCommand.cs

```diff
@@ -94,19 +94,24 @@ public override TestResult Execute(TestExecutionContext context)
         {
             try
             {
-                var testExecution = RunTestOnSeparateThread(context);
-                if (Task.WaitAny(new Task[] { testExecution }, _timeout) != -1
-                    || _debugger.IsAttached)
+                var separateContext = new TestExecutionContext(context)
                 {
-                    context.CurrentResult = testExecution.GetAwaiter().GetResult();
-                }
-                else
-                {
-                    string message = $"Test exceeded Timeout value of {_timeout}ms";
+                    CurrentResult = context.CurrentTest.MakeTestResult()
+                };
+                var testExecution = Task.Run(() => innerCommand.Execute(separateContext));
+                var timedOut = Task.WaitAny(new Task[] { testExecution }, _timeout) == -1;
 
+                separateContext.CurrentResult.ApplyOutput(context.CurrentResult);
```

Comment:
Should this not be the other way around?
You want the _separateContext_ to go to the _context_ not the other way around?

Reply:
From memory (which may be flawed) but I think I saw in the debugger that these were getting _written_ to the original context but that it was getting overridden when we create the testResult from the separateContext (which is what would then get output).

I'll load this up and rerun it soon to confirm

Reply:
I see the method wording here could be confusing too.
Leaving desired functionally aside for a moment, if the intention of this code were to take the output from `separateContext` and apply it to `context` perhaps the method name would be clearer if it were `ApplyOutputTo`?

Reply:
`CopyOutputTo` would be clearer. At the moment unless I got it wrong it does a `CopyOutputFrom`

Reply:
Of course, thanks! That's also more consistent with some BCL methods like `Stream.CopyTo()`